### DISCUSSION TITLE
How is tolerance used in motor.control.target_tolerances(speed, position)?

### DISCUSSION THREAD

**Discussion by deopark** [pybricks/support#1997] [Q&A]
I thought the position parameter was the margin of error for determining if the motor had reached its target.
So I wrote and tested the following code:

```python
from pybricks.hubs import PrimeHub
from pybricks.parameters import Direction, Port, Stop
from pybricks.pupdevices import Motor
from pybricks.tools import StopWatch, multitask, run_task, wait

# Set up all devices.
prime_hub = PrimeHub()
watch = StopWatch()
motor = Motor(Port.B, Direction.CLOCKWISE)

# Initialize variables.
degs = 500
speed = 1000

async def drive():
    await wait(1)
    await motor.run_angle(speed, degs)

async def write_log():
    await wait(1)
    while True:
        await wait(1)
        print('angle : ', motor.angle(), ' speed', motor.speed(), 'time', watch.time())
        await wait(100)

async def main():
    motor.control.target_tolerances(50, 200) ###########################
    motor.reset_angle(0)
    await multitask(drive(), write_log(), race=True)
    print('end=========')

run_task(main())
```

I held the motor by hand and rotated it at the lowest speed, and it stopped at around 300 degrees. (degs 500 - tolerances position 200)

However, if you change the speed, result is completely different.

For example, if test the same way with speed = 500, it will stop at around 430 degrees.

I think I misunderstood the position parameter of target_tolerances().
Can someone explain it to me?

It would be better if there is an experimental example.
thank you.

**Comment by dlech** [comment]
I'm not able to reproduce this experiment. The motor is always going to close to 500 degrees for me. I tried with both stable and beta and with medium and large motors.

**Reply by deopark** [reply]
Thank you.

If so, could you explain the position parameter in target_tolerance (speed, position) in more detail?

Help Docs is not enough for me.

**Comment by laurensvalk** [comment]
> I thought the position parameter was the margin of error for determining if the motor had reached its target.

It is. It will try to get there normally, but "stop trying" if it takes too long but it is within the tolerances.

This is to ensure it won't block your program from proceeding if something is blocking the motor.

In your experiments, it looks like your rotation amount is 500 and your tolerance is 200. In both of your experiments it reached the goal within the tolerance, right?

**Reply by deopark** [reply]
thank you.

I'm focusing on the position which is the olerances parameter.
To trigger the situation, I interfered with the rotation of the motor by holding it with my hand.

However, the position at which the motor stops (the position at which the function ends) appeared differently depending on the speed value.

When target_tolerances (50, 200), if the speed of the motor is 1000, the motor is stopped near 200 degrees short of the target, but if the speed is small(ex, speed=500), it will proceed closer to the target.

In other words, if it set the speed lower, it is not stop even if it comes within the tolerances position. Before it is much close to destination position.

I don't know what kind of regularity there is..

Do tolerances postion means roughly within, not near within?
